repository/release.go

```diff
@@ -2,36 +2,56 @@ package repository
 
 import (
 	"context"
+	"errors"
 
+	"release-manager/pkg/dberrors"
 	"release-manager/repository/model"
-	"release-manager/repository/util"
+	"release-manager/repository/query"
 	svcmodel "release-manager/service/model"
 
-	"github.com/nedpals/supabase-go"
+	"github.com/georgysavva/scany/v2/pgxscan"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
 )
 
-const releaseDBEntity = "releases"
-
 type ReleaseRepository struct {
-	client *supabase.Client
+	dbpool *pgxpool.Pool
 }
 
-func NewReleaseRepository(c *supabase.Client) *ReleaseRepository {
+func NewReleaseRepository(pool *pgxpool.Pool) *ReleaseRepository {
 	return &ReleaseRepository{
-		client: c,
+		dbpool: pool,
 	}
 }
 
 func (r *ReleaseRepository) Create(ctx context.Context, rls svcmodel.Release) error {
-	data := model.ToRelease(rls)
+	if _, err := r.dbpool.Exec(ctx, query.CreateRelease, pgx.NamedArgs{
+		"id":           rls.ID,
+		"projectID":    rls.ProjectID,
+		"releaseTitle": rls.ReleaseTitle,
+		"releaseNotes": rls.ReleaseNotes,
+		"createdBy":    rls.AuthorUserID,
+		"createdAt":    rls.CreatedAt,
+		"updatedAt":    rls.UpdatedAt,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReleaseRepository) Read(ctx context.Context, id uuid.UUID) (svcmodel.Release, error) {
```

Comment:
Added a `Read` function to demonstrate how data will be retrieved from the database. This function is not currently used in the service  at the moment (but it will be of course).

I have also checked [the article](https://www.strv.com/blog/creating-a-project-template-go-with-database-storage). I will start with very simple implementation and helper functions will be added gradually.